src/app/dataset-view/additional-components/dataset-settings-component/tabs/access/dataset-settings-access-tab/dataset-collaborations.service.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright Kamu Data, Inc. and contributors. All rights reserved.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the LICENSE file.
+ */
+
+import { inject, Injectable } from "@angular/core";
+import { ToastrService } from "ngx-toastr";
+import { map, Observable } from "rxjs";
+import { DatasetCollaborationApi } from "src/app/api/dataset-collaboration.api";
+import {
+    AccountWithRoleConnection,
+    DatasetAccessRole,
+    DatasetListCollaboratorsQuery,
+    DatasetSearchCollaboratorQuery,
+    LookupFilters,
+    NameLookupResult,
+    SetRoleCollaboratorMutation,
+    UnsetRoleCollaboratorMutation,
+} from "src/app/api/kamu.graphql.interface";
+
+@Injectable({
+    providedIn: "root",
+})
+export class DatasetCollaborationsService {
+    private datasetCollaborationApi = inject(DatasetCollaborationApi);
+    private toastrService = inject(ToastrService);
+
+    public listCollaborators(params: {
+        datasetId: string;
+        page?: number;
+        perPage?: number;
+    }): Observable<AccountWithRoleConnection> {
+        return this.datasetCollaborationApi
+            .listCollaborators(params)
+            .pipe(
+                map(
+                    (result: DatasetListCollaboratorsQuery) =>
+                        result.datasets.byId?.collaboration.accountRoles as AccountWithRoleConnection,
+                ),
+            );
+    }
+
+    public searchCollaborator(params: {
+        query: string;
+        filters: LookupFilters;
+        page: number;
+        perPage: number;
+    }): Observable<NameLookupResult[]> {
+        return this.datasetCollaborationApi
+            .searchCollaborator(params)
+            .pipe(
+                map((result: DatasetSearchCollaboratorQuery) => result.search.nameLookup.nodes as NameLookupResult[]),
+            );
+    }
```

Comment:
Similarly, no relation to datasets. It's a part of search facilities

Reply:
Fixed